File: app/src/main/java/com/kongpf8848/dmail/login/oauth/OAuthConfiguration.java
package com.kongpf8848.dmail.login.oauth;

import com.kongpf8848.dmail.login.DMAccountType;

import java.util.List;

public interface OAuthConfiguration {

    DMAccountType getAccountType();

    String getClientId();

    String getClientSecret();

    String getAuthorizationURL();

    String getTokenURL();

    String getRedirectURL();

    List<String> getScopes();

}

File: app/src/main/java/com/kongpf8848/dmail/login/oauth/OAuthConfiguration.kt
package com.kongpf8848.dmail.login.oauth

import com.kongpf8848.dmail.login.DMAccountType

interface OAuthConfiguration {
    val accountType: DMAccountType
    val clientId: String
    val clientSecret: String?
    val authorizationURL: String
    val tokenURL: String
    val redirectURL: String
    val scopes: List<String>
}